pyvolcans/pyvolcans_func.py

```diff
@@ -617,6 +645,133 @@ def match_name(volcano_name):
     return matched_volcanoes
 
 
+def plot_bar_apriori_analogues(my_volcano_name, my_volcano_vnum,
+                               my_apriori_analogues, volcans_result,
+                               criteria_weights_text, save_figure=None):
+    """
+    Plots values of single-criterion and total analogy between the target
+    volcano and any a priori analogue provided by the user. It optionally
+    saves the figure in png format (600 dpi resolution).
+
+    Parameters
+    ----------
+    my_volcano_name : str
+        Name of volcano introduced by the user (i.e. target volcano).
+    my_volcano_vnum: int
+        Volcano number (VNUM) of volcano introduced by the user in the
+        GVP database (www.volcano.si.edu).
+    my_apriori_analogues:
+        List of a priori analogues as introduced by the user via the command
+        line.
+    volcans_result : Pandas dataframe
+        Total and single-criterion analogy values between the target volcano
+        and any volcano listed in the GVP database (v. 4.6.7).
+    criteria_weights_text : str
+        Single string indicating the weighting scheme selected by the user.
+    save_figure : bool, optional
+        Keyword argument that indicates whether the generated figures are to
+        be saved in the current working directory, as indicated by the optional
+        flag `--save_figure` chosen by the user when running PyVOLCANS.
+
+    Returns
+    -------
+    my_apriori_analogues_plot : matplotlib.axes.Axes object
+        Instance returned mostly for testing purposes on the function.
+    """
+
+    # derive the indices for all a priori analogues
+    my_apriori_volcano_idx = [convert_to_idx(x) for x in my_apriori_analogues]
+
+    # slice volcans_result to derive a data frame with the a priori analogues
+    all_my_apriori_analogies = \
+        volcans_result.loc[my_apriori_volcano_idx,
+                          ['name','ATs','AG','AM','ASz','ASt']]
+
+    # plot single- and total-analogy values for all a priori analogues
+    my_apriori_analogues_plot = \
+        all_my_apriori_analogies.plot.bar(x="name",
+                                          y=["ATs","AG","AM","ASz","ASt"],
+                                          stacked=True)
```

Comment:
Hi @mobiuscreek!

Many thanks for your comments. Could this code above help solve [this issue](https://github.com/BritishGeologicalSurvey/pyvolcans/pull/5#discussion_r619695803)?

All the best